src/Battlescape/BattlescapeGame.cpp

```diff
@@ -95,7 +95,7 @@ void BattleActionCost::clearTU()
  */
 bool BattleActionCost::haveTU(std::string *message)
 {
-	if (Time <= 0)
+	if (Time <= 0 && Mana <= 0)
```

Comment:
All imaginable actions must cost at least 1 TU. Xcom cannot move or act faster than light. Please remove this sub-feature.

Reply:
Does this also apply to action which are not actually carried out?
There are several possible features I could think of which would justify a 0 TU action game wise. When someone is not acting in that moment, for example.

Reply:
1/ Example please... what soldier skill?
2/ Battleactions are performed on items, not on soldiers... if we want some sort of soldier toggles, they should be handled by a completely new GUI, clicking on the soldier instead of a weapon

Reply:
1) One example would be "Enter Concealment" allowing the soldier to return to the status before being spotted. Another one would be "Run & Gun" which explicitly refreshed the TU for a soldier so he can do another turn, needing one TU to activate that one is just inconvenient. "Reaper" Skill activation:
A devastating chain melee attack, each melee kill in Reaper mode grants an extra action, but further melee attacks deal reduced damage.

That are just three of the skills I have implemented or plan to implement for the ranger class. Each of these could be justifiably activated with 0 TU.

2) That's why I was eager to have special weapons on soldier types. Because the special weapon icon effectively does what I want.

Reply:
OK, then this is definitely a misuse. Special weapon is also just a weapon/item. Soldier skill activation should not be done via item actions. It needs to be a separate context menu (for example by clicking on the soldier rank/portrait or on some extra icon) that indicates it is a soldier skill.

Reply:
What about an icon at the same location of the special weapon icon, that could pop up such a menu? (I have a matching icon for my mod in use.)
If that would be possible, I would not use a weapon/item of course.

Reply:
I would propose two different features then:
1. Scripted Items which align with your guidelines and always have TU costs.
2. A soldier skill menu which can activate actions that do not neccessarily have TU costs.
Would you approve of that?

Reply:
preliminary yes, final yes after I see it